openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -101,7 +101,13 @@ def send_notifications(user_ids, course_key: str, app_name, notification_type, c
     sender_id = context.pop('sender_id', None)
     default_web_config = get_default_values_of_preference(app_name, notification_type).get('web', False)
     generated_notification_audience = []
+
     for batch_user_ids in get_list_in_batches(user_ids, batch_size):
+        if ENABLE_NOTIFICATIONS_FILTERS.is_enabled(course_key):
+            logger.info(f'Sending notifications to {len(batch_user_ids)} users.')
+            batch_user_ids = NotificationFilter().apply_filters(batch_user_ids, course_key, notification_type)
+            logger.info(f'After applying filters, sending notifications to {len(batch_user_ids)} users.')
```

Comment:
can we add `course_key` to all the logs, it would help us!

Reply:
Addded